### DISCUSSION TITLE
Getting the Pico-8 executable to work, is it possible?

### DISCUSSION THREAD

**Discussion by justgage** [OnionUI/Onion#1093] [Q&A] [answered]
Onion OS for the Miyoo Mini Plus seems to be based on Linux so I decided to try to get Pico-8 proper to run on it since FAKE-8 tends to break on a lot of games, plus I miss SPLORE, which seems possible to use given it's WIFI chip. But sadly I'm running into errors.

I coppied the executable from raspberry pi / linux (32bit and 64bit) and they had this error:

```
/mnt/SDCARD/pico-8 2 # ./pico8
./pico8: line 1: syntax error: unexpected word (expecting ")")
```

Any idea what could be the problem?  I'm a bit unfamiliar with what exact architecture that the Miyoo mini plus is, maybe it's just not going to be compatible 😢.

**Comment by XK9274** [comment]
How did you copy the files to your MMP? 

PS, they'll need to be compiled for arm as 32bit, if they're not; or the file has become damaged when transferring (usually happens with filezilla unless you change the transfer type) they won't run and you'll get the error you're seeing.

**Reply by justgage** [reply]
@XK9274 I did transfer them over FTP to my Miyoo Mini Plus. I'll have to try via SD card to see if that makes a difference. They say they're 32 bit but I'm not sure if they are ARM or not, although it _seems_ like they would be? Maybe that's the problem. Maybe I should try again with the Raspberry Pi version.

**Reply by XK9274** [reply]
The error is most likely filezilla but i can't say for sure pico-8 will run for you once transferred correctly

It's a reproduceable issue with FZ and bftpd where if you don't set:

![image](https://github.com/OnionUI/Onion/assets/47260768/014e9a24-d09b-4129-96db-62b25025a2e5)

You'll get errors like the below (and the one you have):

![image](https://github.com/OnionUI/Onion/assets/47260768/da446dc3-52fb-4089-8ef9-25928cc9d05f)

On bins that would otherwise work when transferred as Binary type (or via SFTP/Samba/SD Card):

![image](https://github.com/OnionUI/Onion/assets/47260768/89639eba-e9a1-4788-a338-e4a35b09a69d)

**Comment by courgi** [comment]
Weeks ago I too played around with getting SPLORE to run but was not successful. 
It has been successfully run on several RK3326 handheld devices. 
A reference video: https://www.youtube.com/watch?v=BxED85l9wvo&ab_channel=RetroGameCorps

**Comment by Abssenta** [comment]
Hi. I have tried with the raspberry executable, and I get this message.

`/mnt/SDCARD/pico8 # ./pico8
./pico8: error while loading shared libraries: libbcm_host.so: cannot open shared object file: No such file or directory`

I cannot manually add the missing library to /lib since the mount is ro and it cannot be mounted back as rw. And I don't even think it would work since this lib is part of the raspberry firmware. 

Have anyone succeeded running the original pico executable in the miyoo mini plus?

Regards.

**Comment by gn-dframe** [comment]
Pico-8 uses SDL2 which is not bundled with the Mini's OS. Apparently SDL2 does not work well on the device because it's much more GPU focussed, but I know don't how that would affect the software renderer (if Pico-8 is even using that).

It's possible to build SDL2 using the Miyoo Mini toolchain, so the RPi build of Pico-8 *might* work if `libSDL2.so` could somehow be installed and dynamically linked using the `pico8_dyn` executable. Who knows how the controls would work though.

FWIW this is all a little frustrating because the Mini+ feels like it should be a good candidate to run the RPi Pico-8 given its ARM-based architecture, Ports system and and Wi-fi capabilities, especially since moderately-specced Anbernic devices can run it (albeit on more fully-featured Linux distros such as JELOS).

**Comment by Liquidream** [comment]
I too am still holding out hope for the day when I can use my "original" PICO-8 binaries on the Miyoo Mini Plus 😊🤞

**Comment by ejleslie** [comment]
Just chiming in here to subscribe and add my voice to the chorus of hopefuls. True PICO-8 on the Miyoo Mini Plus (as a port or however it could best be accomplished) would be a huge boon to the system, and one that would fit it so well.

**Comment by XK9274** [comment, marked as answer]
https://www.youtube.com/shorts/xyxIT7lfsyM

You're in luck! 

It has some issues currently that need working out.

**Reply by ejleslie** [reply]
Oh man oh man 😃 any idea when we can expect to see this in the wild?

**Reply by XK9274** [reply]
Very soon i hope.... there's some issues I'm struggling with though. 

It's working in its current state, plays games fine. Has minor audio issues, either has crackling or a delay (similar to drastics problems at the start).

**Reply by gn-dframe** [reply]
Excellent!

**Reply by khromov** [reply]
@XK9274 Amazing, really looking forward to this. Let us know if we can help you beta test it somehow!

**Reply by jan-wir** [reply]
Nice! Also looking forward for this!

**Reply by XK9274** [reply]
I've got to a point where it's completely usable with excellent performance & we have a group of people together for testing but integration hasn't got a timescale yet. Soon™

- Audio is fixed
- Overclocking support & hotkeys (and underclocking ofc)
- Some border overlay support currently

You'll need to provide your own bins (pico8_dyn and the file pico8.dat)

![script_020](https://github.com/OnionUI/Onion/assets/47260768/51dbefcd-c657-4338-951b-fe0d45ef3217)

**Reply by wgrusznis** [reply]
this is amazing work.
thank you for working on this.

**Reply by Liquidream** [reply]
Awesome work @XK9274 - exciting times! 🤓
(\*Raises hand to volunteer for testing\*... 🙋‍♂️)

**Reply by ejleslie** [reply]
Spectacular. Can't wait. ❤️

**Reply by XK9274** [reply]
https://www.reddit.com/r/MiyooMini/comments/17nw79r/announcing_native_pico8_wrapper_for_onion_with/

**Reply by gn-dframe** [reply]
Great job @XK9274! Steward's SDL2 port has really opened the floodgates for this device.

**Reply by XK9274** [reply]
It has yeah, the base version is a great canvas for Pico as it's hardly got any of the drastic stuff in so adding all the other features is really easy, overlays and overclocking went in quite problem free once I figured out the flickering.. audio was the biggest challenge. 😁

It could lead to a lot more aswell, I guess stay tuned.

**Reply by wgrusznis** [reply]
@XK9274 I can't put away my miyoo mini + now, I'm having a blast with pico8 on this little device, with splore running on it it's just to good to be true! Once again thank you for the amazing work

**Reply by courgi** [reply]
A big thank you as well! Pico-8 splore working flawlessly adding endless content. If that wasnt good enough, Drastic NDS came out of nowhere and is truly spectacular. I had no idea the Miyoo mini can come anywhere close to the performance its giving, props to the dev's of course.

With all of the strives in development here lately is there any possibility of Daphne-emu improvements?
